### PR TITLE
feat(#6): add dsn builder

### PR DESCRIPTION
**Pull request by Nightapes**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ This library uses the standard Golang [SQL driver interface](https://golang.org/
 
 ### Create Connection
 
+#### with exasol dsn
```
Comment (AnastasiiaSergienko) [inline README.md:11]:
```suggestion
#### With Exasol DSN 
```

File: README.md [hunk 2]
```diff
@@ -23,6 +25,23 @@ func main() {
 }
 ```
 
+#### with exasol config
```
Comment (AnastasiiaSergienko) [inline README.md:28]:
```suggestion
#### With Exasol Config
```

File: README.md [hunk 3]
```diff
@@ -23,6 +25,23 @@ func main() {
...
+package main
+
+import (
+	"database/sql"
+	
+	"github.com/exasol/exasol-driver-go"
+)
+
+func main() {
+	db, err := sql.Open("exasol", exasol.NewConfig("username", "password").Port(8563).String())
```
Comment (AnastasiiaSergienko) [inline README.md:40]:
```suggestion
	exasol, err := sql.Open("exasol", exasol.NewConfig("<username>", "<password>").Port(<port>).String())
```

Comment (AnastasiiaSergienko) [reply README.md:40]: Would be nice to add Host to the example too

File: integration_test.go [hunk 4]
```diff
@@ -31,31 +33,34 @@ func (suite *IntegrationTestSuite) SetupSuite() {
 }
 
 func (suite *IntegrationTestSuite) TestConnect() {
-	exasol, _ := sql.Open("exasol", "exa:localhost:"+suite.port+";user=sys;password=exasol;encryption=0")
-	rows, _ := exasol.Query("SELECT 2 FROM DUAL")
+	if testing.Short() {
+		suite.T().Skip("skipping integration test")
+	}
```
Comment (AnastasiiaSergienko) [inline integration_test.go:38]: Should it be in `TestIntegrationSuite`? Does it skip only one test method now?

File: integration_test.go [hunk 5]
```diff
@@ -137,7 +142,7 @@ func (suite *IntegrationTestSuite) TestExecuteAndQueryWithContext() {
 }
 
 func (suite *IntegrationTestSuite) TestBeginWithCancelledContext() {
-	exasol, _ := sql.Open("exasol", "exa:localhost:"+suite.port+";user=sys;password=exasol;encryption=0;autocommit=0")
+	exasol, _ := sql.Open("exasol", exasol.NewConfig("sys", "exasol").Insecure(true).Port(suite.port).Autocommit(false).String())
```
Comment (AnastasiiaSergienko) [inline integration_test.go:145]: Should we still have at least one test without a builder?

**Review by AnastasiiaSergienko — CHANGES_REQUESTED** [review]
The builder looks nice. I added a few comments

File: integration_test.go (hunk 5 above)
Comment (Nightapes) [reply integration_test.go:145]: I will add one 👍

File: README.md [hunk 6]
```diff
@@ -8,6 +8,8 @@ This library uses the standard Golang [SQL driver interface](https://golang.org/
 
 ### Create Connection
 
+#### With exasol dsn
```
Comment (AnastasiiaSergienko) [inline README.md:11]: Would be nice to follow the same markdown conventions across the whole documentation. I suggest using the following rules:

Use title case for section headings, preferably the APA style variant:

Capitalize the first word of any heading, including title, subtitle, subheading.
Capitalize all "major" words (nouns, verbs, adjectives, adverbs, and pronouns) in any heading, including the second part of hyphenated major words (e.g., Self-Report not Self-report).
Capitalize all words of five letters or more.
Effectively, only "minor" words of four letters or fewer, namely, conjunctions (words like and, or, nor, and but), articles (the words a, an, and the), and prepositions (words like as, at, by, for, from, in, of, on, per, to, with), are lowercased in any heading, as long as they aren't the first word.

Comment (AnastasiiaSergienko) [reply README.md:11]: This is a rule we use in all public repositories our team maintaining. According to this rule, the header should look like this: `With Exasol DSN`

File: README.md [hunk 7]
```diff
@@ -23,6 +25,23 @@ func main() {
 }
 ```
 
+#### With exasol config
```
Comment (AnastasiiaSergienko) [inline README.md:28]: The same as above

File: README.md [hunk 8]
```diff
@@ -23,6 +25,23 @@ func main() {
...
+package main
+
+import (
+	"database/sql"
+	
+	"github.com/exasol/exasol-driver-go"
+)
+
+func main() {
+	db, err := sql.Open("exasol", exasol.NewConfig("<username>", "<password>").Port(<port>).Host("<host>").String())
```
Comment (AnastasiiaSergienko) [inline README.md:40]: The example would look better if we are consistent in naming. Either we use `exasol` as a variable name everywhere or we pick another name and change it in all other examples. I would suggest avoiding unnecessary abbreviations, but `database` also sounds fine to me

File: integration_test.go [hunk 9]
```diff
@@ -3,59 +3,80 @@ package exasol_test
...
+	rows, _ := db.Query("SELECT 2 FROM DUAL")
 	columns, _ := rows.Columns()
 	suite.Equal("2", columns[0])
 }
 
 func (suite *IntegrationTestSuite) TestConnectWithWrongPort() {
-	exasol, _ := sql.Open("exasol", "exa:localhost:1234;user=sys;password=exasol;encryption=0")
+	if testing.Short() {
+		suite.T().Skip("skipping integration test")
+	}
```
Comment (AnastasiiaSergienko) [inline integration_test.go:52]: Have you tested if we really need this in each method when we place it once in `TestIntegrationSuite`? I thought it was a main entry point into the test

File: README.md (hunk 8 above)
Comment (Nightapes) [reply README.md:40]: I will change everything to `database`, `exasol` is not possible, because of the import

**Review by AnastasiiaSergienko — APPROVED** [review]

**Pull request merged**